src/v2/System/Router/RouterLink.tsx

```diff
@@ -24,40 +24,40 @@ export type RouterLinkProps = Omit<
     noUnderline?: boolean
   }
 
-export const RouterLink: React.ForwardRefExoticComponent<RouterLinkProps> = React.forwardRef(
-  ({ to, noUnderline, ...rest }, forwardedRef) => {
-    const context = useContext(RouterContext)
-    const routes = context?.router?.matcher?.routeConfig ?? []
-    const matcher = context?.router?.matcher
-    const isSupportedInRouter = useMemo(
-      () => !!matcher?.matchRoutes(routes, to),
-      [matcher, routes, to]
-    )
-
-    // TODO: Bulk replace
-    const deprecated = noUnderline ? { textDecoration: "none" } : {}
+export const RouterLink: React.ForwardRefExoticComponent<
+  RouterLinkProps & { ref?: React.Ref<HTMLAnchorElement> }
```

Comment:
I guess we should actually remove this `forwardRef` — since it's unsupported in router aware links.